File: C#/atividades/atividade5/FormasGeometricas/FormasGeometricas/Models/Triangulo.cs
namespace FormasGeometricas.FormaGenerica;


internal class Triangulo : FormaGenerica, ICalculoArea
{
    public float Altura { get; set; }
    public float Base { get; set; }
    public Triangulo(string nome, float altura, float base1) : base(nome)
    {
        Altura = altura;
        Base = base1;
    }

    public override void ExibirInfo()
    {
        base.ExibirInfo();
        CalcularArea();
        CalcularPerimetro();   
    }

    public void CalcularArea()
    {
        Console.WriteLine($"Área do {Nome}: {Altura * Base}");
    }
    public void CalcularPerimetro()
    {
        Console.WriteLine($"Perimetro do {Nome}: {(Altura * 2) + (Base * 2)}");
    }
}

File: JAVA/ORIENTACAO-A-OBJETO/ATIVIDADES/orientacao-a-objeto/video2/atividade  4 ok/FormasGeometricas/src/Triangulo.java
public class Triangulo extends FormaGeometrica implements ICalculavel {
    private double Altura;
    private double Base;
    private double Lado;

    public Triangulo(String cor, String nome, double altura, double base, double lado) {
        super(cor, nome);

        this.Altura = altura;
        this.Base = base;
        this.Lado = lado;
    }

    public double getAltura() {
        return Altura;
    }

    public void setAltura(double altura) {
        Altura = altura;
    }

    public double getBase() {
        return Base;
    }

    public void setBase(double base) {
        Base = base;
    }

    public double getLado() {
        return Lado;
    }

    public void setLado(double lado) {
        Lado = lado;
    }


    @Override
    public void calcularArea() {
        System.out.println("Área do triangulo: " + (this.Base * this.Altura) / 2);
    }

    @Override
    public void calcularPerimetro() {
        System.out.println("Área do triangulo: " + (this.Base + (Lado * Lado)));
    }
}
